feat: add C implementation for `stats/base/dists/exponential/mgf` (#4171)
stdlib-js/stdlib: lib/node_modules/@stdlib/stats/base/dists/exponential/mgf/README.md

<section class="references">

<!-- C interface documentation. -->

* * *

<section class="c">

## C APIs

<!-- Section to include introductory text. Make sure to keep an empty line after the intro `section` element and another before the `/section` close. -->

<section class="intro">

</section>

<!-- /.intro -->

<!-- C usage documentation. -->

<section class="usage">

### Usage

```c
#include "stdlib/stats/base/dists/exponential/mgf.h"
```

#### stdlib_base_dists_exponential_mgf( t, lambda )

> Evaluating the logarithm of the [moment-generating function][mgf] of an [exponential][exponential-distribution] distribution with parameter `lambda`(rate parameter).

Reviewer: ```suggestion
Evaluates the moment-generating function ([MGF][mgf]) for an [exponential][exponential-distribution] distribution.
```
